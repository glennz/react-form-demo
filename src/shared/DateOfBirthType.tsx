import ControlDataType from "./ControlDataType";

type DateOfBirthType = {
    day: ControlDataType,
    month: ControlDataType,
    year: ControlDataType
};

export default DateOfBirthType;