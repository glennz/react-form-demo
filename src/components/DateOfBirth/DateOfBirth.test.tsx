import React from 'react';
import { mount } from 'enzyme';

import { DateOfBirth } from './index';
import DateOfBirthType from '../../shared/DateOfBirthType';



describe('<DateOfBirth />', () => {
    const fieldProps = {
        id: 'test',
        name: 'test',
        label: 'Event date',
        dateOfBirth: {
            day: {value: ''}, 
            month: {value: ''}, 
            year: {value: ''}
        } as DateOfBirthType,
    };

  it('should render a DateOfBirth', () => {
    const dob = mount(<DateOfBirth {...fieldProps}></DateOfBirth>);
    expect(dob.find('div[className="dob-inputs"]')).toHaveLength(1);
    expect(dob.find('input[className="dob-input-field"]')).toHaveLength(3);
    expect(dob.find('.error-message')).toHaveLength(0);
    expect(dob).toMatchSnapshot();
  });

  it('should render a DateOfBirth', () => {
    const dob = mount(<DateOfBirth {...fieldProps} touched={true} error="Date of birth is required" />);
    expect(dob.find('div[className="dob-inputs"]')).toHaveLength(1);
    expect(dob.find('input[className="dob-input-field"]')).toHaveLength(3);
    expect(dob.find('.error-message')).toHaveLength(1);
    const err = dob.find('div[className="error-message"]');
    expect(err.text()).toBe('Date of birth is required');
    expect(dob).toMatchSnapshot();
  });
});