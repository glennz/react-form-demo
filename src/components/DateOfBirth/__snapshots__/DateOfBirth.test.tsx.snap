// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DateOfBirth /> should render a DateOfBirth 1`] = `ReactWrapper {}`;

exports[`<DateOfBirth /> should render a DateOfBirth 2`] = `ReactWrapper {}`;
